Automattic/jetpack
# Boost: Add a "Fix" button to the Image Size Analyzer records to fix image display problems

This PR adds a "Fix" button to each record in the Image Size Analyzer report. When clicked it saves a description of the correct dimensions of the image, and will apply those fixes when the post is displayed.

Fixes https://github.com/Automattic/boost-cloud/issues/371

## Proposed changes:
* Changes the "Fix on Page" button in Image Stabilizer Reports to "Fix".
* Clicking that button will insert a postmeta record with the proper dimensions of the image. It's keyed by the MD5 of the image name or the attachment ID of the image.
* Displaying a post with a fixed image will cause the HTML img tag of any fixed images to be modified so it loads the correct image through Photon, using srcset and sizes attributes.

### Other information:


## Jetpack product discussion
pc9hqz-27e-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Apply patch.
* run `npm run dev` in projects/plugins/boost/
* Edit wp-config.php and add `define( 'ISA_FIX_BUTTON', true );`
* Run an Image Size Analysis of the site and view report.
* Click on any record to view the details of an image problem.
* Click on the "Fix" button.
* Go into PHPMyAdmin to check if a new postmeta record was created in the fixed post. It's called "jb_image_fixes".
* Reload the page and verify that the image is fixed and matches the width of the content area.

Note: wp-config.php is in tools/docker/wordpress/ if you're using a local install.
